sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py

```diff
@@ -1352,7 +1352,7 @@ def _generate_delete_blobs_subrequest_options(
         return query_parameters, header_parameters
 
     def _generate_delete_blobs_options(
-            self, *blobs: Union[str, Dict[str, Any], BlobProperties],
+            self, *blobs: Union[str, Dict[str, Any]],
```

Comment:
I wonder if we should actually keep this for `delete_blobs` as it is sort of a special case. I guess I wonder if we were already pulling out all the information from the properties below (line 1377+). Does a `.get` work on a properties object? If so, we were already taking snapshot, version, etc. from the properties which I think makes more sense in terms of `delete_blobs`.

Curious on @annatisch's thoughts.

Reply:
Yes, I'm also curious about whether we were fully honoring the `BlobProperties` or not - i.e. using `snapshot` and `version`.... if not, then it was already problematic.

Reply:
I think because `BlobProperties` is a `DictMixin` we are actually respecting snapshot, version, etc. here.
https://github.com/Azure/azure-sdk-for-python/blob/674698cd5048b5625713d73406c99681df585803/sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py#L1379-L1392

Given that, do we want to continue supporting `BlobProperties` in these batch APIs (this and `set_blob_tiers` below)? I would argue this is a special case since you need some way to specify that information for each blob in your list. WE have the `Dict[str, Any]` which also works but I think it may make sense to just pass `BlobProperties` if you have them.

Reply:
Let's keep BlobProperties as an input for batch operations (delete_blobs, set_blob_tiers) for now.